app/controllers/users/messages_controller.rb

```diff
@@ -26,7 +26,7 @@ def contact
   private
 
   def set_recipient
-    @recipient_user = User.find(params[:id])
+    @recipient_user = User.find_by(url_name: params[:url_name])
```

Comment:
The behaviour is different here:

```ruby
User.find(91919191919)
# ActiveRecord::RecordNotFound: Couldn't find User with an out of range value for 'id'
# from /home/vagrant/bundle/ruby/2.3.0/gems/activerecord-4.2.10/lib/active_record/core.rb:159:in `rescue in find'

User.find_by(url_name: '91919191919')
# => nil
```

This leads to a 500 error (`undefined method `name' for nil:NilClass`) instead of a 404 if a `url_name` is mistyped.

Reply:
Fixed in bc37e877f16b39f8a01b820b2e760cfd05465c7a (with new spec to check for the expected behaviour)